src/backend/distributed/relay/relay_event_utility.c

```diff
@@ -536,6 +547,70 @@ AppendShardIdToConstraintName(AlterTableCmd *command, uint64 shardId)
 }
 
 
+static bool
+ExtendRowReferencesWalker(Node *node, RowRefWalkerState *state)
+{
+	if (node == NULL)
+	{
+		return false;
+	}
+
+	if (IsA(node, IndexElem))
```

Comment:
> As written, this'll result in the expr being processed twice

Nevermind, just realized this code'll mean `raw_expression_tree_walker` is never called on an `IndexElem`, so it doesn't matter that `9.5.4` and higher know to traverse its `expr` field. This is fine as-is.